packages/mobile/src/identity/feelessVerification.ts

```diff
@@ -0,0 +1,1061 @@
+    Logger.debug(TAG, '@fetchKomenciReadiness', 'Too  many errors')
+    throw new KomenciErrorQuotaExceeded()
+  }
+
+  yield put(
+    feelessUpdateVerificationState({
+      ...feelessVerificationState,
+      komenci: {
+        ...feelessVerificationState.komenci,
+        serviceAvailable: true,
+      },
+    })
+  )
+}
+
+function* fetchPhoneHashDetailsFromCache(e164Number: string) {
+  Logger.debug(TAG, '@fetchPhoneHashDetailsFromCache', 'Starting fetch')
+  const [feelessVerificationState, pepperCache]: [
+    FeelessVerificationState,
+    E164NumberToSaltType
+  ] = yield all([select(feelessVerificationStateSelector), select(e164NumberToSaltSelector)])
+
+  let { phoneHashDetails } = feelessVerificationState
+  let { phoneHash } = feelessVerificationState.phoneHashDetails
+  const ownPepper = pepperCache[e164Number]
+
+  // If we already know the phoneHash, we can bail without updating state
+  if (phoneHash.length) {
+    return phoneHashDetails
+  }
+
+  // Pepper will be queried as part of feeless verification flow but cache
+  // may be empty if the user hasn't initiated a Komenci session before
+  if (!ownPepper) {
+    Logger.debug(TAG, '@fetchPhoneHashDetailsFromCache', 'Pepper not cached')
+    throw new PepperNotCachedError()
+  }
+
+  phoneHash = getPhoneHash(e164Number, ownPepper)
+  phoneHashDetails = { e164Number, phoneHash, pepper: ownPepper }
+
+  yield put(
+    feelessUpdateVerificationState({
+      ...feelessVerificationState,
+      phoneHashDetails,
+    })
+  )
+  return phoneHashDetails
+}
+
+function* fetchSessionState(komenciKit: KomenciKit, e164Number: string) {
+  Logger.debug(TAG, '@fetchSessionState', 'Starting fetch')
+  const [feelessVerificationState, pepperCache, sessionStatusResult]: [
+    FeelessVerificationState,
+    E164NumberToSaltType,
+    Result<CheckSessionResp, FetchError>
+  ] = yield all([
+    select(feelessVerificationStateSelector),
+    select(e164NumberToSaltSelector),
+    call([komenciKit, komenciKit.checkSession]),
+  ])
+
+  let sessionActive = true
+  let {
+    requestQuotaRemaining,
+    pepperQuotaRemaining,
+    unverifiedMtwAddress,
+  } = feelessVerificationState.komenci
+
+  // An inactive session is not fatal, it just means we will need to start one
+  if (!sessionStatusResult.ok) {
+    Logger.debug(TAG, '@fetchSessionState', 'No active sessions')
+    sessionActive = false
+  } else {
+    Logger.debug(TAG, '@fetchSessionState', 'Active session found')
+    const {
+      quotaLeft: { distributedBlindedPepper, requestSubsidisedAttestation, submitMetaTransaction },
+      metaTxWalletAddress,
+    } = sessionStatusResult.result
+    const ownPepper = pepperCache[e164Number]
+
+    Logger.debug(
+      TAG,
+      '@fetchSessionState Session status:',
+      JSON.stringify(sessionStatusResult.result)
+    )
+
+    sessionActive = true
+    // Somtimes `metaTxWalletAddress` is returned as undefined for an active session.
```

Comment:
> Also, in which cases will we have an unverifiedMtwAddress cached but have metaTxWalletAddress be undefined?

If a user starts Session1, has a MTW deployed for them during Session1, then needs to start Session2 because they've run out of quota on Session1. In this case, the `mtwAddress` associated with Session2 is `undefined`.

In this case, it is important that we keep using the `mtwAddress` created in Session1 because attempting to deploy a new MTW in Session2 while using the same EOA as Session1 will cause a Komenci failure. @bowd can provide more background on this specific behavior.


> Separately, why do we have the notion of unverified vs verified metatxwallet address? 

I think my naming may be causing some confusion. I presume any mtw address to be an `unverifiedMtwAddress` until proven otherwise. Once we know an address to be verified, the flow immediately ends so there isn't really any notion of operating on a verified MTW address in this flow.